lib/opal/rewriters/logical_operator_assignment.rb

```diff
@@ -0,0 +1,199 @@
+require 'opal/rewriters/base'
+
+module Opal
+  module Rewriters
+    class LogicalOperatorAssignment < Base
+      class AbstractHandler < self
+        def self.get(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.set(lhs, rhs)
+          raise NotImplementedError
+        end
+
+        def self.process(lhs, rhs, root_type)
+          s(root_type,
+            get(lhs, rhs),
+            set(lhs, rhs))
+        end
+
+        def self.new_temp
+          @@counter ||= 0
+          @@counter += 1
+          :"$or_equals_lhs_recvr_tmp_#{@@counter}"
```

Comment:
> I was implementing cvars in Opal, I trust them 😄

Clearly I had old info 😄👍🏼

> What do you think about extracting and reusing the code that generates temporary variables from the compiler?

Can it be a node type (IIRC there's already `:js_tmp`) with a progressing value as child that acts as an id, then the compiler can have its table of tmpvars and return the same jsid for the same numeric id. eg. 

```
node = s(:temporary_var, 23132)
compiler.get_temp_name(node) # => "TMP_12"
```

similarly to what (I think) happens with symbols in MRI

---

Anyway I'm in favor of the extraction